### ISSUE TITLE
Better bash integration to work with CircleCI and other CI tooling

### ISSUE THREAD

**Issue by justinjmoses** [graphprotocol/graph-cli#351]
I'm setting up continuous integration via CircleCI. I've run into a few issues as the `graph` CLI tool seems to circumvent `bash` standards. 

1. When `graph deploy` fails, such as with invalid auth token, the process exits with error code `0`. It must be something non-zero. 

2. The output logged to screen isn't captureable.  I.e. if you run `graph deploy ... > /dev/null`, you still see all the output (minus the first lines echoing the access token for anyone to see 😳). If you try to `tee` the output, that is also truncated.

You can see these reproduced on this build here: https://circleci.com/gh/Synthetixio/synthetix-subgraph/22 (last step: `Deploy with auth`). This build a) should have failed with a non-zero exit code b) should not have echoed any output, and c) `tee` should have captured the full output.

**Comment by leoyvens** [comment]
On the redirection issue, that's probably because the messages are being printed to stderr, not stdout. Would redirecting both work for you?

**Comment by justinjmoses** [comment]
From the link I posted you can see some of the output (screenshot below) - it doesn't look like error output to me - am I missing something?

![image](https://user-images.githubusercontent.com/799038/67598563-d4fd1e00-f73b-11e9-8ee3-d04c843eac4d.png)

**Comment by leoyvens** [comment]
For example if you want to redirect all output, you need to use `&> /dev/null` which will also redirect stderr, `> /dev/null` will redirect only stdout.

**Comment by justinjmoses** [comment]
Yeah I get that - my question is why would `Apply migrations` for instance be emitted on `stderr`?

**Comment by leoyvens** [comment]
That's a good question, possibly it's the default for gluegun

It seems we don't have a convention right now, a useful one might be to emit the logs to stderr and only the subgraph hash to stdout, in case the hash is useful as input for some other script. Or maybe we should use only stdout to make redirecting the output easier.

**Comment by justinjmoses** [comment]
I think the latter for sure - only use `stderrr` for the actual errors.
